src/py-opentimelineio/opentimelineio/core/_core_utils.py

```diff
@@ -388,3 +394,57 @@ def __deepcopy__(self, *args, **kwargs):
 @add_method(SerializableObject)
 def __copy__(self, *args, **kwargs):
     raise ValueError("SerializableObjects may not be shallow copied.")
+
+
+@add_method(AnyDictionary)
+def to_dict(self):  # noqa: F811
```

Comment:
I'm not convinced that this is the best name. Should we instead call it `to_py`? It would be more generic. I don't like that we have `to_dict` on everything except `AnyVector` which has `to_list`.

Reply:
`to_json`?

Reply:
There is no type named json in Python. `to_json` to me means "to JSON formatted string".

Reply:
OK I see what you mean. `to_py` is indeed more generic then, even if it's a bit less intuitive?

Reply:
I'd prefer `to_dict` and `to_list` over `to_py`.
That way you know what you get. 
Of course the nature of the object will indicate what `to_py` returns, but I still prefer the explicitness of `to_dict` and `to_list`

Reply:
I wouldn't mind using `to_dict` because I'm used to seeing and using this to make a complex structure a simple dictionary. To me it makes sense.

Reply:
`to_py` would allow us to iterate through a list of OTIO objects (of any type) and convert them to python types without using `isinstance` to pick the right method.

Reply:
Follow up to this, I ended up having to make a helper function to do exactly this at work. I ended up calling the function `pythonify` to communicate its returning Python objects.

`to_py` seems in line with other conversion method names on OTIO objects and is clear of what to expect. Another reason to use a `py`/`python` name is because `AnyDictionary` and `AnyVector` can nest each other so you're not just getting a `dict` or `list` but rather the entire struct is `pythonify`-ed.